kolibri/plugins/pdf_viewer/assets/src/views/PdfRendererIndex.vue

```diff
@@ -328,16 +338,22 @@
     },
     mounted() {
       // Retrieve the document and its corresponding object
-      this.prepComponentData.then(() => {
-        // Progress is NaN if loadingProgress.total is undefined
-        if (isNaN(this.progress)) {
-          this.progress = 1;
-        }
-        this.$emit('startTracking');
-        this.updateContentStateInterval = setInterval(this.updateProgress, 30000);
-        // Even if user does not pause while scrolling on first page, we store that as visited
-        this.storeVisitedPage(1);
-      });
+      this.prepComponentData
+        .then(() => {
+          // Progress is NaN if loadingProgress.total is undefined
+          if (isNaN(this.progress)) {
+            this.progress = 1;
+          }
+          this.$emit('startTracking');
+          this.updateContentStateInterval = setInterval(this.updateProgress, 30000);
+          // Even if user does not pause while scrolling on first page, we store that as visited
+          this.storeVisitedPage(1);
+        })
+        .catch(() => {
+          // We have already handled this error above, but we reraised it to ensure that we don't
+          // enter the then block above.
+          // We need to catch it here to avoid an unhandled promise rejection.
```

Comment:
is this part for ie11 specifically, or are there other scenarios where we might encounter this? I understand it was for ie 11 in the context of this issue, but just curious if there are other scenarios where this might be useful

Reply:
Any error during PDF parsing would get to this point - so it could happen from a badly formed PDF, I am pretty sure.